piccolo/columns/reverse_lookup.py

```diff
@@ -0,0 +1,226 @@
+from __future__ import annotations
+
+import inspect
+import typing as t
+from dataclasses import dataclass
+
+from piccolo.columns.base import Selectable
+from piccolo.columns.column_types import (
+    JSON,
+    JSONB,
+    Column,
+    LazyTableReference,
+)
+
+if t.TYPE_CHECKING:  # pragma: no cover
+    from piccolo.table import Table
+
+
+class ReverseLookupSelect(Selectable):
+    """
+    This is a subquery used within a select to fetch reverse lookup data.
+    """
+
+    def __init__(
+        self,
+        *columns: Column,
+        reverse_lookup: ReverseLookup,
+        as_list: bool = False,
+        load_json: bool = False,
+        table: t.Type[Table],
+    ):
+        """
+        :param columns:
+            Which columns to include from the related table.
+        :param as_list:
+            If a single column is provided, and ``as_list`` is ``True`` a
+            flattened list will be returned, rather than a list of objects.
+        :param load_json:
+            If ``True``, any JSON strings are loaded as Python objects.
+        :param table:
+            Parent table for reverse lookup.
+
+        """
+        self.as_list = as_list
+        self.columns = columns
+        self.reverse_lookup = reverse_lookup
+        self.load_json = load_json
+        self.table = table
+
+        safe_types = [int, str]
+
+        # If the columns can be serialised / deserialise as JSON, then we
+        # can fetch the data all in one go.
+        self.serialisation_safe = all(
+            (column.__class__.value_type in safe_types)
+            and (type(column) not in (JSON, JSONB))
+            for column in columns
+        )
+
+    @property
+    def foreign_key_columns_index(self) -> int:
+        reverse_lookup_table = (
+            self.reverse_lookup._meta.resolved_reverse_joining_table
+        )
+        fk_columns = [
+            i._meta.name
+            for i in reverse_lookup_table._meta.foreign_key_columns
+        ]
+        return fk_columns.index(self.table._meta.tablename)
+
+    def get_select_string(self, engine_type: str, with_alias=True) -> str:
+        reverse_lookup_table = (
+            self.reverse_lookup._meta.resolved_reverse_joining_table
+        )
+        reverse_lookup_table_name = reverse_lookup_table._meta.tablename
+        reverse_lookup_pk = reverse_lookup_table._meta.primary_key._meta.name
+        reverse_lookup_fk_table_name = (
+            reverse_lookup_table._meta.foreign_key_columns[
+                self.foreign_key_columns_index
+            ]._meta.name
```

Comment:
I think instead of `_meta.name` it should be `_meta.table._meta.tablename`. Otherwise `reverse_lookup_fk_table_name` is a column name